storage/rocksdb/ha_rockspart.h

```diff
@@ -24,12 +24,11 @@ class ha_rockspart : public native_part::Partition_base {
   ha_rockspart(handlerton *hton, TABLE_SHARE *table_arg)
       : native_part::Partition_base(hton, table_arg){};
 
-  ha_rockspart(handlerton *hton, TABLE_SHARE *share,
-               partition_info *part_info_arg,
-               native_part::Partition_base *clone_arg,
-               MEM_ROOT *clone_mem_root_arg)
-      : native_part::Partition_base(hton, share, part_info_arg, clone_arg,
-                                    clone_mem_root_arg) {}
+  ha_rockspart(handlerton *hton, TABLE_SHARE *table_arg,
+               partition_info *part_info)
+      : native_part::Partition_base(hton, table_arg, &table_arg->ha_share) {
+    this->get_partition_handler()->set_part_info(part_info, true);
```

Comment:
Shouldn't part_info be passed to Partition_base constructor, like the removed clone constructor did?

Reply:
set_ha_share_ref() is overloaded by Parition_base and made private. Hence this has to be passed to Parition_base. If we can make it protected, we don't need to pass this to Partition_base.